experiments/add_sf_to_df_add_row.R

```diff
@@ -350,6 +390,34 @@ geo_attributes <- function(dat, zoomto = NULL){
     # provide mechanism to return after all done
     observeEvent(input$donebtn, {
 
+      if(grepl(class(df$data$geometry)[[1]], "sfc_GEOMETRY")){
+
+        if (any(st_is_empty(df$data$geometry))) {
+          shinyWidgets::show_alert('Missing Geometry',
+                                   'some features are missing geometry, these must be entered before saving',
+                                   type = 'warning')
+        } else {
+          stopApp({
+
+        geo_type <- lapply(df$data$geometry, FUN = function(x){class(x)[[2]]})
```

Comment:
I really like this idea, are you just trying to avoid adding dplyr as a dependancy. You can achive the same with:

```
df$data %>%
    mutate(geo_type = st_geometry_type(.)) 
```

Reply:
yeah didn't check whether it was a dependency or not, so just went with base... Just looked at mapedit imports and 'dplyr' is on there. I'd much rather use dplyr! I'm all for changing, thoughts?

Reply:
yes lets just go with dplyr, the snippet above should do what you wanted (except for the final split)

Reply:
@joshualerickson the other thing I want add here is the ability to reinsert the list of geometries. I quite liked the workflow, of make some changes, save clicking the done button, then later return to add more or make changes. This would not have worked with multiple geometries anyway so the split to list is still best. My first brief attempt gets all the data into the table and can zoom between existing geom, but it does not display. I will keep working on it.

https://github.com/mrjoh3/mapedit/blob/master/experiments/add_sf_to_df_add_row.R#L80